connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java

```diff
@@ -246,8 +256,29 @@ protected void initChannel(SocketChannel ch) throws Exception {
 
                          InetSocketAddress proxyAddr = new InetSocketAddress(u.getHost(),
                                                                              u.getPort() == -1 ? 8080 : u.getPort());
-                         p.addLast(userName == null ? new HttpProxyHandler(proxyAddr)
-                                                    : new HttpProxyHandler(proxyAddr, userName, password));
+                         ProxyHandler proxy = createProxyHandler(jerseyRequest, proxyAddr, userName, password, connectTimeout);
+                         p.addLast(proxy);
+                     } else {
+                         ProxySelector sel = ProxySelector.getDefault();
+                         for (Proxy proxy: sel.select(requestUri)) {
+                             if (proxy.type().equals(Proxy.Type.HTTP)) {
```

Comment:
I'm checking the class java.net.Proxy and it has a constructor where the type could be null. Does it makes sense to change that line by any of these?:
`proxy.type() == Proxy.Type.HTTP`
or
`Proxy.Type.HTTP.equals(proxy.type())`

Reply:
Updated